enhancements/agent-installer/simplified_cluster_operations.md

```diff
@@ -0,0 +1,425 @@
+   of the target cluster.
+4. The user edits the IRI resource by adding the release image pullspec to the
+   `spec.releases` field.
+5. The user waits until the copy process of the new release image is completed
+   for all the control plane nodes. The progress and status of the task could
+   be monitored via the IRI `status` field.
+6. Once completed, the user upgrades the cluster via the `oc adm upgrade
+   --allow-explicit-upgrade --to-image=<release-image>`
+
+#### Post-upgrade optional steps
+
+1. The user edits the IRI resource by removing the release image pullspec not
+   in use anymore.
+2. The specified release image is removed from all the control plane nodes. The
+   progress and status of the task could be monitored via the IRI `status`
+   field.
+
+### Node expansion
+
+1. The user runs the `oc adm node-image create` command from within the
+   disconnected environment (with optionally some additional node
+   configuration).
+2. The user attaches the newly generate node ISO to the target machine(s),
+   and boots it. The progress and status of the task can be monitored via the
+   `oc adm node-image monitor` command.
+3. The user waits until the CSRs are generated, and manually approves them
+   via the `oc adm certificate approve` command.
+   
+### API Extensions
+
+#### InternalReleaseImage custom resource
+
+The `InternalReleaseImage` is a new singleton custom resource, controlled by
+MCO, used to keep track and manage the OCP release images stored into the
+control planes nodes. In particular, it will be used for:
+
+* Triggering a new release image copy task - as a preliminary upgrade step
+* Delete from disk a release image version not anymore in use - as an optional
+  post-upgrade step
+* Enabled/disable the registry services
+
+This is an example of how the CR will look like:
+
+```
+apiVersion: releases.openshift.io/v1
+kind: InternalReleaseImage
+metadata:
+  name: openshift-releases
+spec:
+  releases:
+    - "quay.io/openshift-release-dev/ocp-release:4.18.0-x86_64"
+    - "quay.io/openshift-release-dev/ocp-release:4.19.0-x86_64"
+  enabled: true
+status:
+  releases:
+    - "quay.io/openshift-release-dev/ocp-release:4.18.0-x86_64"
+    - "quay.io/openshift-release-dev/ocp-release:4.19.0-x86_64"
+  enabled: true
+  conditions:
+    - type: "Available"
+      status: "True"
+      lastTransitionTime: "2025-07-12T14:02:00Z"
+      reason: "ReleasesSynchronized"
+      message: "The previously managed releases are available."
+    - type: "Progressing"
+      status: "True"
+      lastTransitionTime: "2025-07-12T13:13:00Z"
+      reason: "AddingRelease"
+      message: "Adding new release image quay.io/openshift-release-dev/ocp-release:4.19.0-x86_64 on master-0."
+  ...
+```
+
+* The `spec.releases` will be used to configure the desired release contents.
+  Adding (or removing) an entry from the list will trigger an update on all the
+  control plane nodes.
```

Comment:
Where do the control plane nodes source the images from?

Reply:
I think I meant, how does the control plane know where to copy the images from, I didn't see anywhere in the API where you configure "ISO is loaded to this mount point"

Reply:
This wasn't clear to me either.